cmd/grype/cli/legacy/event_loop.go

```diff
@@ -9,13 +9,14 @@ import (
 	"github.com/wagoodman/go-partybus"
 
 	"github.com/anchore/clio"
+	"github.com/anchore/grype/grype/grypeerr"
 	"github.com/anchore/grype/internal/log"
 )
 
 // eventLoop listens to worker errors (from execution path), worker events (from a partybus subscription), and
 // signal interrupts. Is responsible for handling each event relative to a given UI an to coordinate eventing until
 // an eventual graceful exit.
-func eventLoop(workerErrs <-chan error, signals <-chan os.Signal, subscription *partybus.Subscription, cleanupFn func(), uxs ...clio.UI) error {
+func eventLoop(workerErrs <-chan error, signals <-chan os.Signal, subscription *partybus.Subscription, cleanupFn func(), uxs ...clio.UI) error { //nolint:gocognit
```

Comment:
Sorry for the `nolint`. I didn't see a great way to clean this up.

Reply:
I'm ok with this being `nolint` - it's legacy and meant to be stripped out eventually for the full cli refactor